pallets/compliance-manager/src/lib.rs

```diff
@@ -134,13 +159,16 @@ pub struct ComplianceRequirement {
 
 /// A compliance requirement along with its evaluation result
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
-#[derive(codec::Encode, codec::Decode, Clone, PartialEq, Eq)]
+#[derive(Encode, Decode, Clone, PartialEq, Eq)]
 pub struct ComplianceRequirementResult {
     pub sender_conditions: Vec<ConditionResult>,
     pub receiver_conditions: Vec<ConditionResult>,
-    /// Unique identifier of the compliance requirement
-    pub id: u32,
-    /// Result of this transfer condition's evaluation
+    /// `Some(u32)` refers to requirement set by the asset issuer
+    /// whilst `None` refers to implicit requirements.
+    ///
+    /// Unique identifier of the compliance requirement.
+    pub id: Option<u32>,
```

Comment:
Implicit requirement result should be handled separately from condition result to avoid confusions.

Reply:
I don't have any problem handling them separately but to make consistent and If I remember the correct UI team needs the condition as well in the rpc result as well as its execution result. That is why I make it compatible with the existing rpc result type.

In some way, we are already treating implicit_requirement differently as `id` should be None to that implicit requirement while others have `Some(id)`. I don't see any confusion here.